core/domain/stats_services.py

```diff
@@ -137,26 +138,33 @@ def _update_stats_transactional(
     """
     exp_stats = get_exploration_stats_by_id(exp_id, exp_version)
     if exp_stats is None:
-        raise Exception(
+        logging.error(
             'ExplorationStatsModel id="%s.%s" does not exist' % (
                 exp_id, exp_version))
+        return
```

Comment:
I think this should be OK since we do a validation to check if the stats request is for the latest exploration version before executing the deferred task:

https://github.com/oppia/oppia/blob/e5677ed4b5429d11baa25eefec4f2b261cec7dd1/core/domain/event_services.py#L84-L89

Reply:
No, I don't follow -- what is wrong with raising the exception? It seems like a stronger check to me than merely logging.

Reply:
My initial reasoning was that exception will cause the task to remain in the queue and pollutes the logs. This scenario can happen if (1) an invalid version is provided to the task or (2) if a stats model does not exist. The former scenario is unlikely under normal operation because once control leaves the controller layer (where it is validated), exploration versions can only increment over time so there cannot be a missing version. The latter can happen when a stats model was deleted or never created in the first place.

Some background: The underlying root cause of the original error was that few stats models were missing, this has been fixed but the fix involved hard reverting to an older version of the exploration -- this made the task fail due to scenario (1) because the version passed to the task no longer exists.

Per discussion offline, I understand that we want the errors to show up (even if the frequency is high so that it gets resolved in a timely manner). So I am now wondering if we can just manually clear the task queue for this case (since the error has been fixed). What are your thoughts?

Reply:
Given the scenario that caused this, why don't we just reject payloads that have a version number greater than the current exploration version? The rationale for doing that is that we don't want to accept such payloads anyway and that there isn't any way to legitimately accept them.

Reply:
Got it, I've changed this a bit. PTAL!